src/System/AtomicWrite/Writer/ByteString.hs

```diff
@@ -46,3 +49,35 @@ atomicWriteFileMaybeMode mmode path text =
   tempFileFor path >>= \(tmpPath, h) -> hPutStr h text
                     >> closeAndRename h tmpPath path
                     >> maybeSetFileMode path mmode
+
+-- | Creates or modifies a file atomically on POSIX-compliant
```

Comment:
I think it'd be a better idea to move these new functions to its own module.

I'm thinking that `System.AtomicWriter.Writer.ByteString.Binary` might be a good place (kind of nest, but it's clear the difference with the current text mode)